sparrow-py/pysrc/sparrow_py/_timestream.py

```diff
@@ -175,8 +181,28 @@ def __add__(self, rhs: Union[Timestream, Literal]) -> Timestream:
         -------
         Timestream
             The Timestream resulting from `self + rhs`.
-        """
-        return Timestream._call("add", self, rhs)
+
+        Raises
+        ------
+        ValueError
+            If attempting to perform an addition that is not possible.
+        """
+        if isinstance(rhs, timedelta):
+            # Right now, we can't convert a time delta directly to a scalar value, because
+            # it potentially has multiple components (days, seconds, microseconds).
+            result = self
+            session = self._ffi_expr.session()
+            if rhs.days != 0:
+                days = Timestream._call("days", rhs.days, session = session)
+                result = Timestream._call("add_time", days, result)
+            if rhs.seconds != 0:
+                seconds = Timestream._call("seconds", rhs.seconds, session = session)
+                result = Timestream._call("add_time", seconds, result)
+            if rhs.microseconds != 0:
+                raise ValueError("Cannot add microseconds to a Timestream")
+            return result
+        else:
+            return Timestream._call("add", self, rhs)
```

Comment:
Doesn't this need to use `add_time` if we have a `rhs` that is a time?

Reply:
At this point we're in `else` an don't have a time.